### ISSUE TITLE
Error : libclntsh.so.11.1 : cannot open shared object file.

### ISSUE THREAD

**Issue by ctwardatulsakinah** [oracle/node-oracledb#442]
Hi,

Im using meteor for my project. I got this error when i add the example of connect.js in my project
I've searched the solution somewhere else and found that it was the problem of the path for  ORACLE_LIB_DIR..
I'd try it and I think i set the path correctly..

I set it like this,

export ORACLE_HOME=/home/Cikkim/app/Cikkim/product/11.2.0/dbhome_1
export ORACLE_SID=orcl
export ORACLE_LIB_DIR=$ORACLE_HOME/lib
export OCI_INC_DIR=$ORACLE_HOME/rdbms/public

but it didn't resolve the issue.. I don't know whether my code is wrong or it is because of the path.

Here are the screenshots of the error,
![11](https://cloud.githubusercontent.com/assets/16010353/15663457/e8250e34-272d-11e6-8eb1-21ee69f48746.PNG)

Below are my code used in meteor.

export const db_oracle = {

init : function(params){
		
		this.alias = (params.alias) ? params.alias : '';
		this.dbName = (params.dbName) ? params.dbName : '';
		this.server = (params.server) ? params.server : 'localhost/orcl';
		this.port = (params.port) ? params.port : 1521;
		this.username = (params.username) ? params.username : '';
		this.password = (params.password) ? params.password : '';
		this.conn = null;
		this.connected = false;
		this.sid = (params.sid) ? params.sid : 'orcl';
		
		return this;
	},

connect : function(){

		var oracledb = Meteor.npmRequire('oracledb');
		
		var self = this;
		var connection = null;
		
		//example of connect.js that I followed
		var connData = oracledb.getConnection(
		{
			alias				: self.alias,
			connectString 		: self.server,
			port 				: self.port,
			user 			: self.username,
			password 		: self.password,
			database 			: self.dbName,
			serviceName		: self.sid
		},
		function(err, connection)
		{
			if (err) {
				console.error(err.message);
				return;
			}
			console.log('Connection was successful!');

			connection.release(
				function(err)
				{
					if (err) {
						console.error(err.message);
						return;
					}
				}
			);
		}
		);

}

Maybe you have the solution to my problem. Thank you.

**Comment by cjbj** [comment]
You probably need to set `LD_LIBRARY_PATH`. See the [installation instructions](https://github.com/oracle/node-oracledb/blob/master/INSTALL.md#instoh):

```
export LD_LIBRARY_PATH=$ORACLE_HOME/lib
```

Node-oracledb doesn't look at `ORACLE_LIB_DIR` at install time or at runtime.  I don't know where you found this.  It can use `OCI_LIB_DIR` at install time, but with an `ORACLE_HOME` install you generally don't need to set either `OCI_LIB_DIR` or `OCI_INC_DIR` (on Linux).

You appear to passing a connection string, so you don't need `ORACLE_SID` set.

Bottom line: set `ORACLE_HOME` and `LD_LIBRARY_PATH`.

**Comment by ctwardatulsakinah** [comment]
Thank you its working.
